src/integer/z/to_string.rs

```diff
@@ -118,6 +122,69 @@ impl Z {
 
         Ok(return_str)
     }
+
+    /// Outputs the integer as a [`Vec`] of bytes.
+    /// The inverse function to [`Z::to_bytes`] is [`Z::from_bytes`] for positive numbers including `0`.
+    ///
+    /// WARNING: The bits are returned as they are stored in the memory. For negative numbers,
+    /// this means that `-1` is output as `[255]`.
+    /// For these values, [`Z::from_bytes`] is not inverse to [`Z::to_bytes`],
+    /// as this function can only instantiate positive values.
+    ///
+    /// Returns a [`Vec<u8>`] of bytes representing the integer as it is stored in memory.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::Z;
+    /// let integer = Z::from(257);
+    ///
+    /// let byte_string = integer.to_bytes();
+    ///
+    /// assert_eq!(vec![1, 1], byte_string);
+    /// ```
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let num_bits = unsafe { fmpz_bits(&self.value) } as usize;
+        let num_bytes = num_bits.div_ceil(8);
+        let mut bytes = vec![0u8; num_bytes];
+
+        for (byte, item) in bytes.iter_mut().enumerate() {
+            for bit_of_byte in (0..8usize).rev() {
+                let bit_index = (byte * u8::BITS as usize + bit_of_byte) as u64;
+                let bit_at_index = unsafe { fmpz_tstbit(&self.value, bit_index) };
+
+                *item *= 2;
+                if bit_at_index == 1 {
+                    *item += 1;
+                }
+            }
+        }
+
+        bytes
+    }
+
+    /// Enables conversion to a UTF8-Encoded [`String`] for [`Z`] values.
+    /// The inverse to this function is [`Z::from_utf8`] for valid UTF8-Encodings.
+    ///
+    /// WARNING: Not every byte-sequence forms a valid UTF8-character.
+    /// If this is the case, a [`FromUtf8Error`] will be returned.
+    ///
+    /// Returns the corresponding UTF8-encoded [`String`] or a
+    /// [`FromUtf8Error`] if the byte sequence contains an invalid UTF8-character.
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`FromUtf8Error`] if the integer's byte sequence contains
+    ///     invalid UTF8-characters.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::Z;
+    /// let integer = Z::from(10);
+    ///
+    /// let text: String = integer.to_utf8().unwrap();
+    /// ```
```

Comment:
```suggestion
    /// Returns the corresponding UTF8-encoded [`String`] or a
    /// [`FromUtf8Error`] if the byte sequence contains an invalid UTF8-character.
    ///
    /// # Examples
    /// ```
    /// use qfall_math::integer::Z;
    /// let integer = Z::from(10);
    ///
    /// let text: String = integer.to_utf8().unwrap();
    /// ```
    ///
    /// # Errors and Failures
    /// - Returns a [`FromUtf8Error`] if the integer's byte sequence contains
    ///     invalid UTF8-characters.
```